Model/ModelManager.php

```diff
@@ -376,7 +386,7 @@ public function getNormalizedIdentifier($entity)
         }
 
         // the entities is not managed
-        if (!$entity /*|| !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity) // commented for perfomance concern */) {
+        if (!$entity || !$this->getEntityManager($entity)->getUnitOfWork()->isInIdentityMap($entity)) {
```

Comment:
@greg0ire - possibly, but in my case the entity should be managed... we're using the standard Sonata\AdminBundle\Controller\CRUDController::editAction(). Let me dig a bit more - there must be a reason why the Entity Manager doesn't see the entity.